plugins/samples/error_page_with_traceid/plugin.cc

```diff
@@ -0,0 +1,146 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_error_page_with_traceid]
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_replace.h"
+#include "proxy_wasm_intrinsics.h"
+
+// Custom HTML template for error pages
+constexpr std::string_view kErrorTemplate = R"(
+<html>
+<head>
+  <title>Error {STATUS_CODE}</title>
+  <style>
+    body { font-family: sans-serif; margin: 2rem; }
+    .container { max-width: 800px; margin: 0 auto; }
+    .trace-id { 
+      background-color: #f5f5f5; 
+      padding: 1rem; 
+      font-family: monospace;
+      word-break: break-all;
+      margin-top: 2rem;
+    }
+  </style>
+</head>
+<body>
+  <div class="container">
+    <h1>Error {STATUS_CODE}</h1>
+    <p>We're sorry, something went wrong with your request.</p>
+    
+    <div class="trace-id">
+      <strong>Trace ID:</strong> {TRACE_ID}
+    </div>
+    
+    <p>Please provide this trace ID to support for assistance.</p>
+  </div>
+</body>
+</html>
+)";
+
+class MyHttpContext : public Context {
+ public:
+  explicit MyHttpContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t headers,
+                                       bool end_of_stream) override {
+    // Capture trace ID early from request headers
+    trace_id_ = extractTraceId();
+    LOG_DEBUG(absl::StrCat("Captured trace ID: ", trace_id_));
+    return FilterHeadersStatus::Continue;
+  }
+
+  FilterHeadersStatus onResponseHeaders(uint32_t headers,
+                                        bool end_of_stream) override {
+    const auto status_header = getResponseHeader(":status");
+    if (!status_header) return FilterHeadersStatus::Continue;
+
+    status_code_ = status_header->toString();
+    const int status_code = std::stoi(status_code_);
```

Comment:
Suggest using absl::SimpleAtoi. IIRC std::stoi can throw an exception, which would fail the plugin.